helix-core/src/main/java/org/apache/helix/controller/stages/ManagementModeStage.java

```diff
@@ -66,6 +68,14 @@ public void process(ClusterEvent event) throws Exception {
     HelixDataAccessor accessor = manager.getHelixDataAccessor();
     ManagementControllerDataProvider cache =
         event.getAttribute(AttributeName.ControllerDataProvider.name());
+    CurrentStateOutput currentStateOutput =
+        event.getAttribute(AttributeName.CURRENT_STATE.name());
+    final Map<String, Resource> resourceMap =
+        event.getAttribute(AttributeName.RESOURCES_TO_REBALANCE.name());
+
+    final BestPossibleStateOutput bestPossibleStateOutput =
```

Comment:
I assume this is only used to cancel pending ST. Calculate it here might be confusing because it is very far away from where it is used.

Reply:
Replied in the above thread. I don't think it's that far. It's used in the next stage. This stage prepares required output for the next stage. It makes sense, also aligned with the default/task pipelines.